includes/class-sensei-content-drip.php

```diff
@@ -124,6 +122,9 @@ public function __construct() {
 	 * Set up all hooks and filters.
 	 */
 	public static function init() {
+		$instance = self::instance();
+		add_action( 'init', array( $instance, 'load_localisation' ), 0 );
```

Comment:
This `$instance` dance looks suspect to me, but I can't quite put my finger on it.

In static cases, other plugins I've reviewed have just done `add_action init [ __CLASS__, 'load_localisation' ]`, not sure why we are micro-managing via `$instance` variable here?

Defining `function Sensei_Content_Drip()` inside a class method `init()` is also a strange architetural move I haven't seen done in a while.

Your thoughts?

Reply:
@lkraav We could have changed that single method to be static, but since we were already getting an instance I don't think it is a problem to have it above. We try to minimize unnecessary breaking signature changes to our public methods. For new hooked methods, we are trying to use static methods when we think someone might want to remove them, but didn't want to break the below hooks for people removing them. Singleton instances allow people to easily remove actions.

You're right that defining a function in a method is strange (but allowed), but for people sniffing for that functions existence to see if content drip is active and I didn't want it to exist if dependencies weren't met. I'd prefer we deprecate these global functions.